disnake/app_commands.py

```diff
@@ -100,11 +100,12 @@ def _validate_name(name: str) -> None:
     # used for slash command names and option names
     # see https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-naming
 
-    assert name == name.lower() and re.fullmatch(r"[\w-]{1,32}", name), (
-        f"Slash command or option name '{name}' should be lowercase, "
-        "between 1 and 32 characters long, and only consist of "
-        "these symbols: a-z, 0-9, -, _, and other languages'/scripts' symbols"
-    )
+    if name != name.lower() or not re.fullmatch(r"[\w-]{1,32}", name):
+        raise InvalidArgument(
```

Comment:
IMO, we should raise a TypeError or ValueError and start to remove InvalidArgument, using TypeError or ValueError instead.

Reply:
Further investigation shows that this is called by functions that already raise InvalidArgument, so I understand the use.

Reply:
Further investigation shows that this is called by functions that already raise InvalidArgument, so I understand the use.